source/extensions/http/original_ip_detection/custom_header/config.cc

```diff
@@ -0,0 +1,35 @@
+#include "extensions/http/original_ip_detection/custom_header/config.h"
+
+#include "envoy/extensions/http/original_ip_detection/custom_header/v3/custom_header.pb.h"
+#include "envoy/extensions/http/original_ip_detection/custom_header/v3/custom_header.pb.validate.h"
+#include "envoy/http/original_ip_detection.h"
+#include "envoy/registry/registry.h"
+
+#include "common/config/utility.h"
+
+#include "extensions/http/original_ip_detection/custom_header/custom_header.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace OriginalIPDetection {
+namespace CustomHeader {
+
+Envoy::Http::OriginalIPDetectionSharedPtr
+CustomHeaderIPDetectionFactory::createExtension(const Protobuf::Message& message,
+                                                Server::Configuration::FactoryContext& context) {
+  auto mptr = Envoy::Config::Utility::translateAnyToFactoryConfig(
+      dynamic_cast<const ProtobufWkt::Any&>(message), context.messageValidationVisitor(), *this);
```

Comment:
Typically this would be done in the outside code so it doesn't need to be done in each extension. See other examples.

Reply:
I had it that way initially, but @alyssawilk actually wanted to keep the HCM agnostic of the types and push this into the extensions... But yeah it isn't consistent across extension points.

Reply:
It should be agnostic of types still? It just uses the factory and known types to resolve?

Reply:
yeah but IMO it's nice to have all the validation logic locally.  Maybe worth a helper to avoid cut and paste but it's nice to be able to unit test validation here, rather than having to spin up everything in HCM tests, WDYT?

Reply:
I don't feel strongly about but I don't know of any other extension type that repeats this type of code in each extension.

Reply:
I like the idea of another helper (in a follow-up round) to DRY this up so leaving it for now. It is indeed nice to be able to test out those validations independently of the extension point code (e.g.: the HCM). Thanks to that, I learned about the header validations and we decided to do strict validations. So I think this is a nice pattern (and it'll be less repetitive with a helper).

Reply:
Resolving this for now since there are no strong feelings.